python/ray/dataframe/utils.py

```diff
@@ -282,3 +282,60 @@ def decorator(cls):
         return cls
 
     return decorator
+
+
+def _map(func, *objects):
+    """Maps a function across a list of objects stored in Ray.
+
+    Args:
+        func (callable): function taking 1 argument.
+        objects: list of objects or ObjectIds.
+
+    Returns:
+        List of objects returned by func.
+    """
+    return [_deploy_func.remote(func, obj) for obj in objects]
+
+
+@ray.remote
+def _reduce(func, *sequence):
```

Comment:
I see what you're doing here, but there might be a cheaper/simpler way.

Reply:
I'm not sure (at least for getting dtypes) whether this needs a tree-reduce implementation. Regardless, I think you can make it simpler by performing it iteratively, something like
```python
def _reduce(func, oids):
    while len(oids) > 1:
        oids = [_remote_reduce_helper.remote(func, item1, item2) for item1, item2 in pairwise(oids)]
    return oids[0]
```
You can use `_submit` to perform on more than just a pair at a time, but this way you construct the entire computation graph locally.